src/lib/components/Communities/CommunityHeader.js

```diff
@@ -1,68 +1,84 @@
+    const { changeSelectedCommunity, community, imagePlaceholderLink } =
+      this.props;
 
-  return (
-    <Container className="deposits-community-header" fluid>
-      <Container className="community-header">
-        {community ? (
-          <Image
-            className="community-logo-header"
-            src={communityLogo}
-            fallbackSrc={imagePlaceholderLink}
-          />
-        ) : (
-          <span>{i18next.t('No community selected.')}</span>
-        )}
-
-        <div className="community-header-info">
-          {communityTitle}
-          <CommunitySelectionModal
-            onCommunityChange={setCommunity}
-            chosenCommunity={community}
-            trigger={
-              <Button
-                size="mini"
-                className="community-header-button ml-5"
-                color="blue"
-                name="setting"
-                type="button"
-              >
-                {i18next.t('Edit')}
-              </Button>
-            }
-          />
-        </div>
+    return (
+      <Container className="deposits-community-header" fluid>
+        <Container className="community-header">
+          {community ? (
+            <>
+              <Image
+                className="community-logo-header"
+                src={community.logo || imagePlaceholderLink}
+                fallbackSrc={imagePlaceholderLink}
+              />
+              {community.title || community.uuid}
+            </>
+          ) : (
+            <span>{i18next.t('No community selected.')}</span>
+          )}
+          <div className="community-header-info">
+            <CommunitySelectionModal
+              onCommunityChange={(community) => {
+                changeSelectedCommunity(community);
+              }}
+              chosenCommunity={community}
+              trigger={
+                <Button
+                  size="mini"
+                  className="community-header-button ml-5"
+                  color="blue"
+                  name="setting"
+                  type="button"
+                >
+                  {i18next.t('Edit')}
+                </Button>
+              }
+            />
+          </div>
+        </Container>
       </Container>
-    </Container>
-  );
-};
+    );
+  }
+}
 
 CommunityHeaderComponent.propTypes = {
   imagePlaceholderLink: PropTypes.string.isRequired,
-  community: PropTypes.object.isRequired,
-  setCommunity: PropTypes.func.isRequired,
+  community: PropTypes.shape({
+    uuid: PropTypes.string.isRequired,
+    title: PropTypes.string.isRequired,
+    logo: PropTypes.string.isRequired,
+  }),
 };
+
+const mapStateToProps = (state) => ({
+  community: state.deposit.community.selected,
+});
+
+const mapDispatchToProps = (dispatch) => ({
+  changeSelectedCommunity: (community) =>
+    dispatch(changeSelectedCommunity(community)),
+});
+
+export const CommunityHeader = connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(CommunityHeaderComponent);
```

Comment:
Usually we do this part in a separate index.js file. Wouldn't it be more consistent to also do it here?

Reply:
IMO, it is annoying to have it in another file... I cannot see immediately what `props` are injected...